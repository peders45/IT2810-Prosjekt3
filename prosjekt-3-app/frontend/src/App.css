/*Main app*/
.App {
  display: flex;
  flex-direction: row;
  text-align: center;
  background-color: red;
}

.App-logo {
  height: 40vmin;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #09d3ac;
}

/*Media queries for smaller screens*/
@media screen and (max-width: 1120px) {
  .App{
    display: flex;
    flex-direction: column;
    justify-content: center;
  }
  .Filter {
    order: 1;
  }
  .Display {
    order: 3;
  }
  .DropDown {
    order: 2;
  }
}
