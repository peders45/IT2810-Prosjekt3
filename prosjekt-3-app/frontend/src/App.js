import React from 'react';
import './App.css';
import SearchField from './components/atoms/SearchField/SearchField'
import CategoryCheckbox from './components/atoms/CategoryCheckbox/CategoryCheckbox'
import SortingDropDown from './components/atoms/SortingDropDown/SortingDropDown'

class App extends React.Component{

  render() {
    return (
      <div className="App">
        <SearchField />
        <CategoryCheckbox category="Drinks" />
        <CategoryCheckbox category="Breakfast" />
        <CategoryCheckbox category="Dessert" />
        <SortingDropDown />
      </div>
    );
  }
}

export default App;
