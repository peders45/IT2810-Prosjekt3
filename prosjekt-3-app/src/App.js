import React from 'react';
import './App.css';
import SearchField from './components/atoms/SearchField/SearchField'

class App extends React.Component{

  render() {
    return (
      <div className="App">
        <SearchField/>
      </div>
    );
  }
}

export default App;
