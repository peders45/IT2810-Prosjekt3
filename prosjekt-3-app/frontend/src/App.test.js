import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store'

const middlewares = [thunk]
const mockStore = configureStore(middlewares)

describe('Rendering', () => {
  it('renders without crashing', () => {
    const initialState = { 
      searchWord: "",
      menus: [],
      category: [],
      sliderMaxCalories: 1880,
      sliderRating: 0,
      sortCategory: "",
      sortValue: null,
      status: "",
      first: 9,
      offset: 0,
      count: 0,
      error: []
    }
    const store = mockStore(initialState)

    const div = document.createElement('div');
    ReactDOM.render(
      <Provider store={store}>
        <App />
      </Provider>, div);
  });

  it('renders correctly with no props', () => {
    const component = shallow(<App/>);
    expect(component).toMatchSnapshot();
  });
});