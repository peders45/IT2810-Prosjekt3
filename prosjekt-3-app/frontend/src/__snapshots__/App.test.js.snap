// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering renders correctly with no props 1`] = `ShallowWrapper {}`;
